src/transformers/models/idefics2/processing_idefics2.py

```diff
@@ -285,10 +286,14 @@ def apply_chat_template(
                 chat_template = self.chat_template
             else:
                 chat_template = self.default_chat_template
-
-        return self.tokenizer.apply_chat_template(
-            conversation, chat_template=chat_template, tokenize=tokenize, **kwargs
-        )
+        with warnings.catch_warnings():
+            # TODO Matt: This is a workaround to avoid annoying warnings until we properly remove default
```

Comment:
Can you open an issue so this work is tracked and we don't forget?

Reply:
The short answer is that because the `chat_template` arg can also be the **name** of a template in the template dict, as well as a full template string, we still sniff `tokenizer.default_chat_template` even if we receive a `chat_template` arg, to check if the chat template matches one of the dict keys.

This is definitely suboptimal, but I'll open an issue and clean it up once we can get rid of default chat templates, which have been causing other problems too!

Reply:
eek - that's not good. That also means anyone is going to get this error if they pass in a proper template. This should really be addressed at the moment, rather than waiting two months for the deprecation cycle

Reply:
Yeah, you're right - I was banking on that being an uncommon path, but I'll try to make a proper fix in this PR!

Reply:
@amyeroberts should be fixed now! I moved the warnings out of the default templates and into the `apply_chat_template` logic itself, so it should now only fire when appropriate. This means the `TODO` is no longer necessary either.